# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../wxwidgets24/freetype-fix.patch
# Copyright (C) 2008 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

diff -Nur wxPythonSrc-2.4.2.4-orig/src/generic/dcpsg.cpp wxPythonSrc-2.4.2.4/src/generic/dcpsg.cpp
--- wxPythonSrc-2.4.2.4-orig/src/generic/dcpsg.cpp	2008-08-25 14:58:19.000000000 +0200
+++ wxPythonSrc-2.4.2.4/src/generic/dcpsg.cpp	2008-08-25 14:58:59.000000000 +0200
@@ -1146,8 +1146,13 @@
   FILE *file;
 };
 
+#if (FREETYPE_MAJOR == 2 && FREETYPE_MINOR >= 2) || FREETYPE_MAJOR >= 2
+static int paps_move_to( const FT_Vector* to,
+			 void *user_data)
+#else
 static int paps_move_to( FT_Vector* to,
 			 void *user_data)
+#endif
 {
   OutlineInfo *outline_info = (OutlineInfo*)user_data;
   fprintf(outline_info->file, "%d %d moveto\n",
@@ -1156,8 +1161,13 @@
   return 0;
 }
 
+#if (FREETYPE_MAJOR == 2 && FREETYPE_MINOR >= 2) || FREETYPE_MAJOR >= 2
+static int paps_line_to( const FT_Vector*  to,
+			 void *user_data)
+#else
 static int paps_line_to( FT_Vector*  to,
 			 void *user_data)
+#endif
 {
   OutlineInfo *outline_info = (OutlineInfo*)user_data;
   fprintf(outline_info->file, "%d %d lineto\n",
@@ -1166,9 +1176,15 @@
   return 0;
 }
 
+#if (FREETYPE_MAJOR == 2 && FREETYPE_MINOR >= 2) || FREETYPE_MAJOR >= 2
+static int paps_conic_to( const FT_Vector*  control,
+			  const FT_Vector*  to,
+			  void *user_data)
+#else
 static int paps_conic_to( FT_Vector*  control,
 			  FT_Vector*  to,
 			  void *user_data)
+#endif
 {
   OutlineInfo *outline_info = (OutlineInfo*)user_data;
   fprintf(outline_info->file, "%d %d %d %d conicto\n",
@@ -1179,10 +1195,17 @@
   return 0;
 }
 
+#if (FREETYPE_MAJOR == 2 && FREETYPE_MINOR >= 2) || FREETYPE_MAJOR >= 2
+static int paps_cubic_to( const FT_Vector*  control1,
+			  const FT_Vector*  control2,
+			  const FT_Vector*  to,
+			  void *user_data)
+#else
 static int paps_cubic_to( FT_Vector*  control1,
 			  FT_Vector*  control2,
 			  FT_Vector*  to,
 			  void *user_data)
+#endif
 {
   OutlineInfo *outline_info = (OutlineInfo*)user_data;
   fprintf(outline_info->file,
